pkg/asyncapi/error.go

```diff
@@ -0,0 +1,62 @@
+package asyncapi
+
+import "fmt"
+
+type errType = int
+
+const (
+	errInvalidProperty errType = iota + 1
+	errInvalidDocument
+	errUnsupportedAsyncapiVersion
+)
+
+var (
+	ErrInvalidDocument = &asyncapiError{
+		errType: errInvalidDocument,
+	}
+	ErrUnsupportedAsyncapiVersion = &asyncapiError{
+		errType: errUnsupportedAsyncapiVersion,
+	}
+)
+
+type asyncapiError struct {
+	errType
+	context interface{}
+}
+
+type Error interface {
+	error
+	InvalidProperty() bool
+	InvalidDocument() bool
+	UnsupportedAsyncapiVersion() bool
+}
+
+func (err asyncapiError) Error() string {
+	switch err.errType {
+	case errInvalidProperty:
+		return fmt.Sprintf("asyncapi: error invalid property %v", err.context)
+	case errInvalidDocument:
+		return "asyncapi: unable to decode document"
+	default:
+		return "asyncapi: unsupported asyncapi version"
+	}
+}
+
+func (err asyncapiError) InvalidProperty() bool {
```

Comment:
```suggestion
func (err asyncapiError) IsInvalidProperty() bool {
```

Similar fix for others.